args:
  dtype:
  - tensor
  - tensor
  - str
  is_pos:
  - false
  - false
  - false
  name:
  - data
  - segment_ids
  - name
  required:
  - true
  - true
  - false
name: tf.math.segment_mean
package: null
pass_rate: 30.21978021978022
rules:
- - cot: 'Based on the given runtime information, the error is caused because the
      attribute ''segment_ids'' is a float, but it is expected to be of type int32
      or int64. The tensors ''segment_ids'' is provided as a float, which is not compatible.
      Therefore, the constraint to prevent the error is:'
    length: 1
    target:
      choosen_dtype:
        data: tensor
        name: str
        segment_ids: tensor
      msg: "Value for attr 'Tindices' of float is not in the list of allowed values:\
        \ int32, int64\n\t; NodeDef: {{node SegmentMean}}; Op<name=SegmentMean; signature=data:T,\
        \ segment_ids:Tindices -> output:T; attr=T:type,allowed=[DT_FLOAT, DT_DOUBLE,\
        \ DT_INT32, DT_UINT8, DT_INT16, 10440210506161272279, DT_UINT16, DT_COMPLEX128,\
        \ DT_HALF, DT_UINT32, DT_UINT64]; attr=Tindices:type,allowed=[DT_INT32, DT_INT64]>\
        \ [Op:SegmentMean] name: xoWN"
      package: tensorflow
    txt: dtype(segment_ids) in [int32, int64]
  - f1_score: 100.0
    overall_score: 100
    precision: 100.0
    recall: 100.0
- - cot: divided
    length: 1
    target:
      choosen_dtype:
        data: tensor
        name: str
        segment_ids: tensor
      msg: '{{function_node __wrapped__SegmentMean_device_/job:localhost/replica:0/task:0/device:CPU:0}}
        input must be at least rank 1 [Op:SegmentMean] name: Ltkq'
      package: tensorflow
    txt: (data.rank > 0)
  - f1_score: 100.0
    overall_score: 100
    precision: 100.0
    recall: 100.0
- - cot: 'segment_ids should have a rank of 1. Therefore, the constraint to prevent
      the error is:'
    length: 1
    target:
      choosen_dtype:
        data: tensor
        name: str
        segment_ids: tensor
      msg: '{{function_node __wrapped__SegmentMean_device_/job:localhost/replica:0/task:0/device:CPU:0}}
        segment_ids should be a vector. [Op:SegmentMean] name: qDVB'
      package: tensorflow
    txt: segment_ids.rank == 1
  - f1_score: 100.0
    overall_score: 100
    precision: 100.0
    recall: 100.0
- - cot: 'Based on the runtime information, we can formulate the constraint to prevent
      the error as follows:'
    length: 1
    target:
      choosen_dtype:
        data: tensor
        name: str
        segment_ids: tensor
      msg: '{{function_node __wrapped__SegmentMean_device_/job:localhost/replica:0/task:0/device:CPU:0}}
        segment_ids should be the same size as dimension 0 of input. [Op:SegmentMean]
        name: uflD'
      package: tensorflow
    txt: segment_ids.shape[0] == data.shape[0]
  - f1_score: 100.0
    overall_score: 100
    precision: 100.0
    recall: 100.0
- - cot: default
    length: 2
    target:
      choosen_dtype:
        data: tensor
        name: str
        segment_ids: tensor
      msg: negative dimensions are not allowed
      package: null
    txt: all(i >= 0 for i in data.shape) and all(i >= 0 for i in segment_ids.shape)
  - f1_score: -1
    overall_score: -1
    precision: -1
    recall: -1
- - cot: default
    length: 2
    target:
      choosen_dtype:
        data: tensor
        name: str
        segment_ids: tensor
      msg: Too large tensor shape
      package: null
    txt: data.rank <= 7 and segment_ids.rank <= 7
  - f1_score: -1
    overall_score: -1
    precision: -1
    recall: -1
